galaxy_importer/collection.py

```diff
@@ -77,6 +84,23 @@ def _import_collection(file, filename, logger, cfg):
     return attr.asdict(data)
 
 
+def _extract_tar_shell(filepath, extract_dir):
+    cwd = os.path.dirname(os.path.abspath(filepath))
+    file_name = os.path.basename(filepath)
+    args = [
+        'tar',
+        f'--directory={extract_dir}',
+        '-xf',
+        file_name,
+    ]
+    try:
```

Comment:
I'd suggest doing the try/excepts in _import_collection  so _extract_tar_shell stays semi generic.

Handling the exceptions in _import_collection also gives you more context to construct exceptions and error messages.